app/src/main/java/com/android/sample/ui/authentication/AvailabilityScreen.kt

```diff
@@ -0,0 +1,174 @@
+
+              // Display the availability grid
+              AvailabilityGrid(
+                  schedule = profile.schedule,
+                  onScheduleChange = { updatedSchedule ->
+                    val updatedProfile = profile.copy(schedule = updatedSchedule)
+                    onProfileUpdate(updatedProfile)
+                  })
+
+              Spacer(modifier = Modifier.weight(1f))
+
+              Button(
+                  onClick = { /* Add your logic for finding a student */
+                    profile.schedule.forEachIndexed { dayIndex, daySchedule ->
+                      println("Day $dayIndex: ${daySchedule.joinToString()}")
+                    }
+                  },
+                  modifier = Modifier.fillMaxWidth(),
+                  colors = ButtonDefaults.buttonColors(Color(0xFF9B59B6))) {
+                    Text(text = "Let's find a student!", color = Color.White)
+                  }
+            }
+      }
+}
+
+@Composable
+fun AvailabilityGrid(schedule: List<List<Int>>, onScheduleChange: (List<List<Int>>) -> Unit) {
+  val days = listOf("Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun")
+  val hours = (8..19).map { "$it h" }
+
+  // We need to copy the schedule locally
+  var selectedSlots by remember { mutableStateOf(schedule) }
+
+  // Grid Layout
+  Column {
+    Row {
+      Spacer(modifier = Modifier.width(44.dp)) // Placeholder to align with time slots
+      days.forEach { day ->
+        Box(
+            modifier =
+                Modifier.weight(1f)
+                    .padding(4.dp)
+                    .background(
+                        Color(0xFFE0E0E0), shape = RoundedCornerShape(8.dp)) // Box for day labels
+                    .padding(4.dp)) {
+              Text(
+                  text = day,
+                  fontSize = 14.sp,
+                  textAlign = TextAlign.Center // Center text inside the box
+                  )
+            }
+      }
+    }
+
+    hours.forEachIndexed { hourIndex, hour ->
+      Row(verticalAlignment = Alignment.CenterVertically) {
+        Box(
+            modifier =
+                Modifier.offset(y = (-18).dp)
+                    .padding(4.dp)
+                    .background(
+                        Color(0xFFE0E0E0), shape = RoundedCornerShape(8.dp)) // Box for hour labels
+                    .padding(4.dp)
+                    .weight(1f)) {
+              Text(
+                  text = hour,
+                  fontSize = 14.sp,
+                  textAlign = TextAlign.Center // Center text inside the box
+                  )
+            }
+
+        days.forEachIndexed { dayIndex, _ ->
+          val isSelected = selectedSlots[dayIndex][hourIndex] == 1
+
+          Box(
+              modifier =
+                  Modifier.weight(1f)
+                      .aspectRatio(1f)
+                      .padding(4.dp)
+                      .background(
+                          if (isSelected) Color(0xFF9B59B6) else Color.LightGray,
```

Comment:
we should use the Material.theme colours to improve modularity if we want to change the app palettes. Since design is not that important at this point, we might let @AlexisAllmnd handle this later!